agreement/cryptoVerifier.go

```diff
@@ -205,7 +205,14 @@ func (c *poolCryptoVerifier) voteFillWorker(toBundleWait chan<- bundleFuture) {
 			}
 
 			uv := votereq.message.UnauthenticatedVote
-			c.voteVerifier.verifyVote(votereq.ctx, c.ledger, uv, votereq.TaskIndex, votereq.message, c.votes.out)
+			err := c.voteVerifier.verifyVote(votereq.ctx, c.ledger, uv, votereq.TaskIndex, votereq.message, c.votes.out)
+			if err != nil && c.votes.out != nil {
+				select {
+				case c.votes.out <- asyncVerifyVoteResponse{index: votereq.TaskIndex, err: err}:
+				default:
+					c.log.Infof("poolCryptoVerifier.voteFillWorker unable to write failed enqueue response to output channel")
+				}
```

Comment:
It seems like with this non-blocking send, and the fact that demux.go is listening on c.votes.out in a big select on lots of channels (on d.crypto.VerifiedVotes()) , you will almost always end up in this default case. Is that right? If so maybe not trying to introduce a new type of error response at all is best?

Reply:
This fail case is quite rare: the `c.voteVerifier.verifyVote` would normally return right away after enqueuing the vote verification.

The user of the `cryptoVerifier` expects the crypto verifier to return a response on the `c.votes.out` channel. If the `c.voteVerifier.verifyVote` doesn't return it, the vote "disappear". The change here makes sure that the vote won't disappear without leaving any trace.

Reply:
Right, but if no one is waiting at the right time, the vote will disappear due to the non-blocking send, right? The only thing waiting on c.votes.out is demux, so I was just predicting that pretty often it will miss the chance to receive this anyway.

Reply:
it shouldn't. The output buffer is sized so it won't lose votes.

Reply:
oh, I see, I didn't realize out was a buffered channel. OK!